nova/virt/vmwareapi/vm_util.py

```diff
@@ -1260,6 +1342,9 @@ def get_stats_from_cluster(session, cluster):
         if policy and hasattr(policy, 'failoverHosts'):
             failover_hosts = set(h.value for h in policy.failoverHosts)
 
+        group_ret = prop_dict.get('configurationEx', {}).get('group')
```

Comment:
Here if you fall to the default {} for configurationEx then group_ret will be None. Then calling _get_host_reservations_map with groups being None will result to exception trying to iterate with a for loop over None.

Reply:
True. Found it, too. Fixed it locally already and wrote some tests for `_get_host_reservations_map` including the `None` case.